Source/AI.cs

```diff
@@ -55,25 +62,41 @@ public string SystemPrompt(Persona currentPersona)
 			}, settings);
 
 			return new List<string>
-			{
-				$"You are role-playing a commentator called '{currentPersona.name}'. ",
-				(otherObservers.Any() ? $"Your co-moderators are {otherObservers} and you all" : $"You") + $" are watching '{player}' playing the popular game Rimworld.\n",
-				$"Your input comes from the current game and will be json like this: {exampleInput}\n",
+			{	$"You are {currentPersona.name}\n",
+				$"The narrative needs to fit within the context of Rimworld, your responses should reflect the game's events and setting appropriately..\n",
+				$"Your input comes from the game, and will be json like this: {exampleInput}\n",
 				$"Your output must only be in json like this: {exampleOutput}\n",
 				$"Limit ResponseText to no more than {currentPersona.phraseMaxWordCount} words.\n",
+				$"When constructing the 'ResponseText', consider vocal clarity and pacing so that it is easily understandable when spoken by Microsoft Azure Speech Services.\n",
 				$"Limit NewHistoricalKeyEvents to no more than {currentPersona.historyMaxWordCount} words.\n",
-				$"Your role/personality is: {currentPersona.personality}\n",
-				$"Remember: your output is in the format: {{\"ResponseText\":\"...\",\"NewHistoricalKeyEvents\":[\"...\",\"...\"]}}",
-			}.Join(delimiter: "").ApplyVoiceStyle(currentPersona);
+				$"{currentPersona.personality}\n",				
+				$"Remember: your output is in the format: {{\"ResponseText\":\"Your narrative response goes here, within the word limit.\",\"NewHistoricalKeyEvents\":[\"...\",\"...\"]}}",
+			}.Join(delimiter: "");
+		}
+
+		private string GetCurrentChatGPTModel()
+		{
+			// Switch models based on the configured ratio.
+			if ((_modelSwitchCounter++ % RimGPTMod.Settings.ModelSwitchRatio) == 0)
```

Comment:
I was under the impression that you switch between the different models more intentionally. Like summarizing with 4 and creative content with 3.5. This suggests that you simply switch periodically between the cheap and the expensive model. Which would mean that sometimes the content is more and sometimes its less intelligent. Hmmm.

Reply:
No, I just switch between 4 and 3.5 periodically.  While 3.5 makes dumber messages, usually a period switch to 4 significantly improves 3.5 in future.  I noticed this when playing the game and switching from 4 to 3.5 manually.  So this is just to automate that process.  That way, when 4 injects an intelligent response, it "corrects" the narrative from 3.5 - It's not perfect though, and we could improve it.